# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/wasm/WasmValidate.h
# Commit: 7805ddfe2bfa
# Full Hash: 7805ddfe2bfa83383dc422ed1e8ac70bda627a89
# Author: Ryan Hunt <rhunt@eqrion.net>
# Date: 2020-08-19 10:01:16
# Regressor Bug: 1561521
# File Overlap Count: 2
# Description:
#   Bug 1561521 - Rename TypeCode/ValType/RefType::Any to Extern. r=lth
#   
#   This commit renames the TypeCode/ValType/RefType occurrences of 'any' to 'extern'
#   to match what they actually represent. All public API facing occurrences have
#   already been switched to 'externref' so this should only be aligning the
# ==============================================================================

diff -r 4fa52a0617de -r 7805ddfe2bfa js/src/wasm/WasmValidate.h
--- a/js/src/wasm/WasmValidate.h	Tue Aug 18 16:58:21 2020 +0000
+++ b/js/src/wasm/WasmValidate.h	Tue Aug 18 16:58:32 2020 +0000
@@ -234,7 +234,7 @@
 #if defined(ENABLE_WASM_GC)
     if (gcTypesEnabled()) {
       // Structs are subtypes of AnyRef.
-      if (isStructType(one) && two.isAnyRef()) {
+      if (isStructType(one) && two.isExternRef()) {
         return true;
       }
       // Struct One is a subtype of struct Two if Two is a prefix of One.
@@ -669,7 +669,7 @@
     switch (code) {
       case uint8_t(TypeCode::OptRef):
         return RefType::fromTypeIndex(uncheckedReadVarU32());
-      case uint8_t(TypeCode::AnyRef):
+      case uint8_t(TypeCode::ExternRef):
       case uint8_t(TypeCode::FuncRef):
         return RefType::fromTypeCode(TypeCode(code));
       default:
@@ -701,7 +701,7 @@
 #endif
 #ifdef ENABLE_WASM_REFTYPES
       case uint8_t(TypeCode::FuncRef):
-      case uint8_t(TypeCode::AnyRef):
+      case uint8_t(TypeCode::ExternRef):
         if (!refTypesEnabled) {
           return fail("reference types not enabled");
         }
@@ -750,7 +750,7 @@
     }
     switch (code) {
       case uint8_t(TypeCode::FuncRef):
-      case uint8_t(TypeCode::AnyRef):
+      case uint8_t(TypeCode::ExternRef):
         *type = RefType::fromTypeCode(TypeCode(code));
         return true;
 #ifdef ENABLE_WASM_GC